Mathlib/Analysis/SpecialFunctions/Complex/Arg.lean

```diff
@@ -536,6 +562,41 @@ lemma arg_mul_eq_add_arg_iff {x y : ℂ} (hx₀ : x ≠ 0) (hy₀ : y ≠ 0) :
 
 alias ⟨_, arg_mul⟩ := arg_mul_eq_add_arg_iff
 
+/-! ### Arc-length -/
+
+/-- The arc-length between two complex numbers is the absolute value of their argument.
+
+The arc-length to `0` is always zero. -/
+noncomputable def arcLength (x y : ℂ) : ℝ := |(x / y).arg|
```

Comment:
Should we be using `euclidean_geometry.angle` somehow here?

Reply:
I'm not sure that `arcLength` is a good name. For me, "arclength" means "arclength of a curve".

Reply:
It is definitely referred to as "arc-length" in the additive combinatorics literature when `x` and `y` are unit complex numbers. However as Eric points out my new definition is a special case of `InneProductGeometry.angle`, so this is moot.

I will record here that my definition actually differed from `InnerProductGeometry.angle` in that it had junk value `0` when `x` or `y` is zero while `InnerProductGeometry.angle` has junk value `π / 2`. It turns out that I'm wrong and Joseph Myers is right, because his junk value means that the triangle inequality for the angle is true (but not yet proved) even if one of the three points is `0`.

Reply:
For points on the unit circle, it is the arc length of the shortest arc of the unit circle that joins the points. I think that we should add a type (not a `Subtype`) for a sphere and use this metric there (at least in the case of an inner product space). However, for `x = 2` and `y = 3 * I`, it has nothing to do with arc length of a curve.

Reply:
Yes, I wrote all of this in my original PR description, but I edited it out since.

Note that we do not have the triangle inequality for `angle`, so we currently can't prove `MetricSpace unitSphere`.

Reply:
Could you please adjust the PR title?